bfdd/bfd_packet.c

```diff
@@ -1257,3 +1567,37 @@ int bp_echov6_socket(const struct vrf *vrf)
 
 	return s;
 }
+
+/* get peer's mac address to be used with Echo packets */
+void bfd_peer_mac_set(int sd, struct bfd_session *bfd,
+		      struct sockaddr_any *peer, struct interface *ifp)
+{
+	struct arpreq arpreq_;
+
+	if (CHECK_FLAG(bfd->flags, BFD_SESS_FLAG_MAC_SET))
+		return;
+
+	/* if BFD session is not up get Peer's Mac Address */
+	if (peer->sa_sin.sin_family == AF_INET) {
+		/* IPV4 */
+		struct sockaddr_in *addr =
+			(struct sockaddr_in *)&arpreq_.arp_pa;
+
+		bzero(&arpreq_, sizeof(struct arpreq));
+		addr->sin_family = AF_INET;
+		memcpy(&addr->sin_addr.s_addr, &peer->sa_sin.sin_addr, 4);
+		strcpy(arpreq_.arp_dev, ifp->name);
+
+		if (ioctl(sd, SIOCGARP, &arpreq_) < 0) {
```

Comment:
This `ioctl` is Linux specific, so this could should either be adapted or made Linux specific.

Reply:
there is an ARP API that has been developed for NHRP.
what about using ZEBRA_NEIGH_DISCOVER ?

Reply:
@pguibert6WIND I'm not familiar with NHRP, but it seems `ZEBRA_NEIGH_DISCOVER` doesn't return anything even as async response. How can we get the peer MAC address this way? Capture the ARP request response?